File: python/cw.py
from plumbum import FG
from plumbum.cmd import grep, wc
import click


@click.command()
@click.argument('word', type=str, required=True)
def main(word):
    (grep["^" + word, "/usr/share/dict/words"]
     | wc["-l"]
     ) & FG


if __name__ == '__main__':
    main()

File: scala/cw.scala
//> using scala 3.5.1

//> using dep com.lihaoyi::os-lib:0.11.3
//> using dep com.lihaoyi::mainargs:0.7.6

import mainargs.{main, arg, ParserForMethods, Flag}

object CW {
  @main
  def run(
    @arg(doc = "A prefix to count words starting with it", positional = true)
    prefix: String
  ) = {
    val grep = os.spawn(cmd = ("grep", "^" + prefix, "/usr/share/dict/words"))
    val wc = os.spawn(cmd = ("wc", "-l"), stdin = grep.stdout, stdout = os.Inherit)
  }

  def main(args: Array[String]): Unit = ParserForMethods(this).runOrExit(args)
}
